### PR TITLE
Load balancing with Envoy

### PR DESCRIPTION
**Pull request by rshriram**

### CODE REVIEW

File: _docs/concepts/load-balancing.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+              
+parent: Traffic Management
+order: 25
+
+bodyclass: docs
+layout: docs
+type: markdown
+---
+
+Istio relies on the platform to automatically register pods/containers to
```
Comment (geeknoid) [inline _docs/concepts/load-balancing.md]: I think we should weaken and generalize this. When Istio is running on bare HW or straight VMs, there is no omnipresent guardian that takes care of this stuff, humans will have to manually provision this kind of state.

File: _docs/concepts/load-balancing.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+re-routed through Envoy. Envoy distributes the traffic across instances in
+the load balancing pool. While Envoy supports several 
+[sophisticated load balancing algorithms](https://lyft.github.io/envoy/docs/intro/arch_overview/load_balancing.html), 
+Istio currently allows three load balancing modes: 
+round robin, random, and weighted least request.
+
+In addition to load balancing, Envoy periodically checks the health of each
+instance in the pool. Envoy follows a circuit breaker style pattern to
+classify instances as unhealthy or healthy based on their failure rates for
+the the health check API call. In other words, when the number of health
```
Comment (geeknoid) [inline _docs/concepts/load-balancing.md]: the the -> the

File: _docs/concepts/load-balancing.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+check failures for a given instance exceeds a pre-specified threshold, it
+will be ejected from the load balancing pool.
```
Comment (geeknoid) [inline _docs/concepts/load-balancing.md]: Maybe add a sentence explaining how the instance can reenter the pool?

File: _docs/concepts/load-balancing.md [hunk 4]
```diff
@@ -28,6 +37,12 @@ round robin, random, and weighted least request.
 In addition to load balancing, Envoy periodically checks the health of each
 instance in the pool. Envoy follows a circuit breaker style pattern to
 classify instances as unhealthy or healthy based on their failure rates for
-the the health check API call. In other words, when the number of health
+the health check API call. In other words, when the number of health
 check failures for a given instance exceeds a pre-specified threshold, it
-will be ejected from the load balancing pool.
+will be ejected from the load balancing pool. Similarly, when the number of
```
Comment (geeknoid) [inline _docs/concepts/load-balancing.md]: Some places you use a dash in load-balancing, some places you don't. Let's pick one and add it to our list of authoring term guidelines that Louis started.

**Comment by rshriram** [comment]
@louiscryan feel free to leave additional review comments. Merging as of now, to prevent big merge conflicts.

**Pull request merged**